packages/gatekeeper/src/helpers/services.ts

```diff
@@ -94,15 +99,14 @@ export async function fetchUser(
 }
 
 /** some docstring */
-export const fetchState = async (
-  url: string,
-  authenticateActionCreator: ActionCreator<AuthenticateAction> = authenticateUser,
-  recordResultActionCreator: ActionCreator<RecordAction> = recordResult,
-  authenticationProgressCreator: ActionCreator<
-    AuthenticationProgressAction
-  > = authenticationProgress,
-  errorCallbackFn: ErrorCallback = errorCallback
-) => {
+export const fetchState = async ({
+  url = '',
+  authenticateActionCreator = authenticateUser,
+  recordResultActionCreator = recordResult,
+  authenticationProgressCreator = authenticationProgress,
+  errorCallbackFn = errorCallback,
+  logoutActionCreator = logOutUser
```

Comment:
I am not sure that I like relying on inferred types Vs statically defining them.  What are your thoughts on this?

Reply:
yeah, inferred types are weaker to statically defined types

Reply:
Maybe we can use a convention where required parameters are passed as ordered arguments and optional arguments are passed in as an object? cc @cKellyDesign.

Reply:
that sounds like a good idea @p-netm --- how would you type such a function?

Reply:
the options would have to be the first argument.

Reply:
why is that @p-netm ?

Reply:
I haven't thought this through totally, i will build a case for it and if compelling enough i will bring it up. At the moment i know it could make it easier in dealing with optional arguments, where we can use a function in modules from which we dont have access to some of the values that would be passed to optional arguments e.g
__contrived-alert__
`const findX(y, z =1, i=2) => {}`

using findX in a module that does not have acces to a value of z, but does for i

Reply:
> why is that @p-netm ?

Not necessarily, I thought about it and the position of the optional arguments as an object is inconsequential to how we modify the existing code.